toc/rfc/rfc-draft-add-support-for-file-based-service-binding.md

```diff
@@ -0,0 +1,123 @@
+
+```
+action := &models.RunAction{
+  Path: "/path/to/executable",
+  Args: []string{"some", "args to", "pass in"},
+  Dir: "/path/to/working/directory",
+  User: "username",
+  EnvironmentVariables: []*models.EnvironmentVariable{
+    {
+      Name: "ENVNAME",
+      Value: "ENVVALUE",
+    },
+  },
+ Files: []*models.Files{
+    {
+      Name: "/etc/cf-instance-binding",
+      Value: "VALUE",
+    },
+  },
+  ResourceLimits: &models.ResourceLimits{
+    Nofile: 1000,
+  },
+  LogSource: "some-log-source",
+  SuppressLogOutput: false,
+}
+```
+
+## Workstreams
+
+### App Runtime Interfaces WG
+
+Cloud Controller should be extended with an App Feature to activate the new file base Service Binding option. If the file-based service binding feature is active the Cloud Controller should generate a Run action, which configures the Service Bindings to be stored as tmpfs file(s) in the application container instead of `VCAP_SERVICES` environment variable.
+
+Additionally, the suggest limit for the size should be implemented.
+
+### App Runtime Platform WG
+
+Diego should add support for the new argument of the Run action to create files with the desired content. Like the [Instance Identity credentials](https://docs.cloudfoundry.org/devguide/deploy-apps/instance-identity.html) implementation, the [Diego Executor](https://github.com/cloudfoundry/executor) should be extended to prepare the `tmpfs` mount and create the required files for an application container. For reference there is a [CredManager](https://github.com/cloudfoundry/executor/blob/db9758c0142ae9c11dad26de672735fb20566105/depot/containerstore/credmanager.go) , InstanceIdentityHandler and the `tmpfs` mount is configured in the [Diego release](https://github.com/cloudfoundry/diego-release/blob/2d7d7c1373f2a61077c74e33a397a5f69b11b131/jobs/rep/templates/setup_mounted_data_dirs.erb#L38-L56) for the current implementation of the Instance Identity Credentials. The files should have security permissions like the Instance Identity Credentilas `644` or even higher `600` owned by the uid running the app if possible.
+
+## Possible Future Work
+
+The App Features API aren’t supported currently in the CF CLI and [app manifest](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html). To make the use of this proposal for CF operators easier this should be addressed.
+
+### App Manifest Attributes Proposal
+
+The CF app manifest is [additive](https://v3-apidocs.cloudfoundry.org/#apply-a-manifest-to-a-space) not declarative. That is why if we want to disable app feature flags via the app manifest a status input like `enabled` or `disabled` will be required. Like:
+
+```
+---
+applications:
+- name: test-app
+  features:
+  -	file-based-service-bindings: true
+```
+or as alternative proposal:
+
+```
+---
+applications:
+- name: test-app
+  features:
+  -	file-based-service-bindings
```

Comment:
"name" is missing before file-based-service-bindings in the alternative proposal:

```
features:
  - name: file-based-service-bindings
    enabled: true
```